src/ServiceManagement/HDInsight/Commands.HDInsight/Cmdlet/SetAzureHDInsightDefaultStorageCmdlet.cs

```diff
@@ -56,30 +56,26 @@ public AzureHDInsightConfig Config
                     throw new ArgumentNullException("value", "The value for the configuration can not be null.");
                 }
 
-                if (value.CoreConfiguration != null)
```

Comment:
Implementation was copied from [AddAzureHDInsightScriptActionCmdlet.cs](https://github.com/Azure/azure-powershell/blob/e06a22c22f7a1c8153e9c2fa7c159ae7b1fd1dab/src/ServiceManagement/HDInsight/Commands.HDInsight/Cmdlet/AddAzureHDInsightScriptActionCmdlet.cs).  These values should reliably be non-null since the [AzureHDInsightConfig constructor](https://github.com/Azure/azure-powershell/blob/dev/src/ServiceManagement/HDInsight/Commands.HDInsight/Model/DataObjects/AzureHDInsightConfig.cs#L28-L46) instantiates each member and the property setters are private.